src/main/java/dev/jlibra/admissioncontrol/AdmissionControl.java

```diff
@@ -0,0 +1,149 @@
+import types.Transaction.RawTransaction;
+import types.Transaction.SignedTransaction;
+import types.Transaction.TransactionArgument;
+import types.Transaction.TransactionArgument.ArgType;
+
+public class AdmissionControl {
+
+    private String host;
+
+    private int port;
+
+    public AdmissionControl(String host, int port) {
+        this.host = host;
+        this.port = port;
+    }
+
+    private static Map<Type, ArgType> jlibraArgumentTypeToGrpcArgumentType;
+
+    static {
+        jlibraArgumentTypeToGrpcArgumentType = new HashMap<>();
+        jlibraArgumentTypeToGrpcArgumentType.put(Type.ADDRESS, ArgType.ADDRESS);
+        jlibraArgumentTypeToGrpcArgumentType.put(Type.U64, ArgType.U64);
+    }
+
+    public SubmitTransactionResult submitTransaction(PublicKey publicKey, PrivateKey privateKey,
+            Transaction transaction) {
+
+        ManagedChannel channel = ManagedChannelBuilder.forAddress(host, port)
+                .usePlaintext()
+                .build();
+
+        AdmissionControlBlockingStub stub = AdmissionControlGrpc.newBlockingStub(channel);
+
+        List<TransactionArgument> transactionArguments = transaction.getProgram().getArguments().stream()
+                .map(txArgument -> TransactionArgument.newBuilder()
+                        .setType(jlibraArgumentTypeToGrpcArgumentType.get(txArgument.type()))
+                        .setData(ByteString.copyFrom(txArgument.toByteArray()))
+                        .build())
+                .collect(toList());
+
+        Program program = Program.newBuilder()
+                .addAllArguments(transactionArguments)
+                .setCode(readCodeFromStream(transaction))
+                .addAllModules(new ArrayList<ByteString>())
+                .build();
+
+        RawTransaction rawTransaction = RawTransaction.newBuilder()
+                .setProgram(program)
+                .setExpirationTime(transaction.getExpirationTime())
+                .setGasUnitPrice(transaction.getGasUnitPrice())
+                .setMaxGasAmount(transaction.getMaxGasAmount())
+                .setSenderAccount(ByteString.copyFrom(KeyUtils.toByteArrayLibraAddress(publicKey.getEncoded())))
+                .setSequenceNumber(transaction.getSequenceNumber())
+                .build();
+
+        SignedTransaction signedTransaction = SignedTransaction.newBuilder()
+                .setRawTxnBytes(rawTransaction.toByteString())
+                .setSenderPublicKey(ByteString.copyFrom(KeyUtils.stripPublicKeyPrefix(publicKey.getEncoded())))
+                .setSenderSignature(ByteString.copyFrom(LibraHelper.signTransaction(rawTransaction, privateKey)))
+                .build();
+
+        SubmitTransactionRequest submitTransactionRequest = SubmitTransactionRequest.newBuilder()
+                .setSignedTxn(signedTransaction)
+                .build();
+
+        SubmitTransactionResponse response = stub.submitTransaction(submitTransactionRequest);
+
+        channel.shutdown();
```

Comment:
I'm not sure if this is the best idea to create and shutdown the channel every time you want to submit transaction. What do you think about having it open on the class level and have methods to start/init AdmissionControl and close it on destroy?

Reply:
Yes, I think this is an open issue overall, but I'm not yet sure how to do this so that it would remain simple to use but be efficient also. Maybe should check how they do similar things in the web3j project.
I will create a separate issue about this.